src/app/App.js

```diff
@@ -1,12 +1,28 @@
-import React from 'react';
-import {
-  Switch,
-  Route,
-} from "react-router-dom";
+import React, { useState, useEffect } from 'react';
+import { Switch, Route, Link } from 'react-router-dom';
+import { auth, createUserProfileDocument } from '../firebase/firebase.utils';
+import Layout from '../layout/Layout';
+import Login from '../components/Login/Login';
 import './App.scss';
-import Layout from "../layout/Layout";
 
 const App = () => {
+  const [currentUser, setCurrentUser] = useState(null);
+
+  useEffect(() => {
+    auth.onAuthStateChanged(async (userAuth) => {
+      if (userAuth) {
+        const userRef = await createUserProfileDocument(userAuth);
+
+        userRef.onSnapshot((snapShot) => {
+          setCurrentUser({
+            id: snapShot.id,
+            ...snapShot.data(),
+          });
+        });
+      }
+      setCurrentUser(userAuth);
+    });
+  }, []);

```

Comment:
to powinno byc trzymane w reduxie zmienna currentUser i metoda która sprawdza czy jestes zalogowany czy nie.

Reply:
moze sama metoda i odpalenie moze byc tak jak jest ale na pewno currentUser nie moze byc trzymany w lokalnym stanie tylko w reduxie

Reply:
tak wiem, docelowo  state bedzie w reduxie, dzisiaj zaczynam.     normalnie bym tego nie commitowal ale  nie zrobilem branach od mojego brancha i wypchnolem to teraz aby  pozniej nie było  zbyd obszernego  materialu o sprawdzeni